### PR TITLE
feat: Move response types to `@empathyco/x-types` package.

### PR DESCRIPTION
**Pull request by javieri-empathy**
EX-6092

This just moves the responses types to the `x-types` package. 

- Discovery wall requests are not moved as they are not standardised yet.
- Original response interfaces are kept in search-adapter package like the requests to prevent breaking changes yet.

### CODE REVIEW

File: packages/search-types/src/response/response.types.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { TaggingRequest } from '../request';
+
+/**
+ * Response to be implemented by all responses that contain a part to track the show event.
+ *
+ * @public
+ */
+export interface TrackableShowResponse {
+  showTagging: TaggingRequest;
+}
```
Comment (mnavarroespinosa) [inline packages/search-types/src/response/response.types.ts:10]: What would you think about renaming this file to `trackeable-show-response.model.ts`?

File: packages/search-types/src/response/identifier-results-response.model.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Result } from '../result';
```
Comment (LuisMartinez15) [inline packages/search-types/src/response/identifier-results-response.model.ts:1]:
```suggestion
import { Result } from '../result/result.model';
```

File: packages/search-types/src/response/next-queries-response.model.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { NextQuery } from '../query-signals';
```
Comment (LuisMartinez15) [inline packages/search-types/src/response/next-queries-response.model.ts:1]:
```suggestion
import { NextQuery } from '../query-signals/next-query.model';
```

File: packages/search-types/src/response/recommendations-response.model.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { Result } from '../result/result.model';
+
+/**
+ * Response for the recommendations' endpoint.
```
Comment (LuisMartinez15) [inline packages/search-types/src/response/recommendations-response.model.ts:4]:
```suggestion
 * Response for the recommendations endpoint.
```

File: packages/search-types/src/response/query-suggestions-response.model.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { Suggestion } from '../suggestion.model';
+
+/**
+ * Response for the suggestions endpoint.
```
Comment (LuisMartinez15) [inline packages/search-types/src/response/query-suggestions-response.model.ts:4]:
```suggestion
 * Response for the query suggestions endpoint.
```

File: packages/search-types/src/response/response.types.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { TaggingRequest } from '../request';
```
Comment (LuisMartinez15) [inline packages/search-types/src/response/response.types.ts:1]:
```suggestion
import { TaggingRequest } from '../request/tagging-request.model';
```

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]
good job!

File: packages/search-types/src/response/response.types.ts (hunk 1 above)
Comment (mnavarroespinosa) [reply packages/search-types/src/response/response.types.ts]: Sure, you're right, that makes sense.

**Review by mnavarroespinosa — APPROVED** [review]
Looks good to me, nice job!

**Review by LuisMartinez15 — APPROVED** [review]

**Review by mnavarroespinosa — APPROVED** [review]

**Pull request merged**